task_management/index.js

```diff
@@ -0,0 +1,333 @@
+import { fetchData, url, handleError } from "./data_file.js";
+
+const addTask = document.querySelector(".addTaskSymbol");
+const addTaskPopupOverlay = document.querySelector(".popupOverlay");
+const addTaskCloseButton = document.querySelector(".closeButton");
+const addTaskButton = document.getElementById("addTaskButton");
+const editTaskPopupOverlay = document.querySelector(".editPopupOverlay");
+const editTaskInput = document.getElementById("editTaskInput");
+const saveEditButton = document.getElementById("saveEditButton");
+const cancelEditButton = document.getElementById("cancelEditButton");
+const taskContainer = document.querySelector(".rightBox");
+const errorMessage = document.getElementById("errorMessage");
+const errorMessageEdit = document.getElementById("errorMessageinedit");
+
+let allTasks = [];
+let currentFilterType = "all";
+
+addTask.addEventListener("click", () => {
+  addTaskPopupOverlay.style.display = "flex";
+});
+
+addTaskCloseButton.addEventListener("click", () => {
+  addTaskPopupOverlay.style.display = "none";
+});
+
+const renderTasksBasedOnFilter = (filterType) => {
+  taskContainer.innerHTML = "";
+
+  const heading = document.createElement("h4");
+  heading.className = "taskHeading";
+  heading.textContent =
+    filterType.charAt(0).toUpperCase() + filterType.slice(1) + " Tasks";
+  taskContainer.appendChild(heading);
+
+  let tasksToDisplay;
+
+  switch (filterType) {
+    case "completed":
+      tasksToDisplay = allTasks.filter((task) => task.isCompleted);
+      break;
+    case "incomplete":
+      tasksToDisplay = allTasks.filter((task) => !task.isCompleted);
+      break;
+    case "favorite":
+      tasksToDisplay = allTasks.filter((task) => task.isFavorite);
+      break;
+    default:
+      tasksToDisplay = allTasks;
+  }
+
+  if (!tasksToDisplay.length) {
+    const noTasksMessage = document.createElement("div");
+    noTasksMessage.textContent = "No tasks available";
+    noTasksMessage.className = "noTasksMessage";
+    taskContainer.appendChild(noTasksMessage);
+  } else {
+    tasksToDisplay.forEach((task) => {
+      const taskElement = createTaskElement(task);
+      taskContainer.appendChild(taskElement.element);
+    });
+  }
+};
+
+const sortTasksByName = (ascending = true) => {
+  allTasks.sort((a, b) => {
+    const nameA = a.name.toLowerCase();
+    const nameB = b.name.toLowerCase();
+
+    if (nameA < nameB) return ascending ? -1 : 1;
+    if (nameA > nameB) return ascending ? 1 : -1;
```

Comment:
Why two conditions are there for same things, does the two condition is doing same thing or different ?

Reply:
find a optimize way to do this logic